pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -112,10 +112,10 @@ assert withCryptsetup ->
 let
   wantCurl = withRemote || withImportd;
 
-  version = "247.6";
 in
-stdenv.mkDerivation {
-  inherit version pname;
+stdenv.mkDerivation rec {
```

Comment:
I don't think this is a positive change at all. We introduce a very large `rec set` here because of what exactly? We are using it at various places within the file and let expressions sounds like the perfect use case for this.

Reply:
We use rec everywhere.

Reply:
This is not a justification for using it here.